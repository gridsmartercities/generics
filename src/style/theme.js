const theme = {
  main: {},
  input: {},
  label: {},
  buttons: {},
  containers: {}
};

export default theme;
