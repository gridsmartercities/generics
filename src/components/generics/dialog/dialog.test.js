import "jest-dom/extend-expect";
import React from "react";
import { Simulate } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import Dialog from "./index";
import theme from "../../../style/theme";
import { ThemeProvider } from "styled-components";

test("Dialog Component", () => {
  const { getByLabelText, container } = render(
    <ThemeProvider theme={theme}>
      <>
        <Dialog id="exercises-open" active={true}>
          Hello
        </Dialog>
        <Dialog id="exercises-closed" active={false}>
          Hello
        </Dialog>
      </>
    </ThemeProvider>
  );
  const dialogOpen = document.querySelector("section#exercises-open");
  const dialogClosed = document.querySelector("section#exercises-closed");
  expect(dialogOpen.dataset.active).toBe("true");
  expect(dialogClosed).toBe(null);

  expect(container.firstChild).toMatchSnapshot();
});
