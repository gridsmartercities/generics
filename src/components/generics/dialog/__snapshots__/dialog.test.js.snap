// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component 1`] = `
<div>
  <div
    class="sc-bdVaJa kNiqos"
  >
    <div
      class="sc-bwzfXH gVxKjR"
    />
    <section
      class="sc-htpNat RGXSK"
      data-active="true"
      id="exercises-open"
    >
      Hello
    </section>
  </div>
</div>
`;
